lib/urlwatch/reporters.py

```diff
@@ -175,48 +166,98 @@ def _parts(self):
             yield SafeHtml('<hr>')
 
         yield SafeHtml("""
-        <address>
-        {pkgname} {version}, {copyright}<br>
-        Website: {url}<br>
-        watched {count} URLs in {duration} seconds
-        </address>
-        </body>
-        </html>
-        """).format(pkgname=urlwatch.pkgname, version=urlwatch.__version__, copyright=urlwatch.__copyright__,
-                    url=urlwatch.__url__, count=len(self.job_states), duration=self.duration.seconds)
-
-    def _diff_to_html(self, unified_diff):
-        for line in unified_diff.splitlines():
-            if line.startswith('+'):
-                yield SafeHtml('<span class="unified_add">{line}</span>').format(line=line)
-            elif line.startswith('-'):
-                yield SafeHtml('<span class="unified_sub">{line}</span>').format(line=line)
-            else:
-                yield SafeHtml('<span class="unified_nor">{line}</span>').format(line=line)
+            <div style="font-family:monospace;font-style:italic">
+            <span>{pkgname} {version}, {copyright}</span>
+            <address>Website: <a href="{url}">{url}</a></address>
+            <span>Watched {count} URLs in {duration} seconds</span>
+            </div>
+            </body>
+            </html>""").format(pkgname=urlwatch.pkgname, version=urlwatch.__version__, copyright=urlwatch.__copyright__,
+                               url=urlwatch.__url__, count=len(self.job_states), duration=self.duration.seconds)
+
+    def _diff_to_html(self, unified_diff, is_markdown=False):
+        if is_markdown and Markdown:
+            # rebuild html from markdown using markdonw2 library's Markdown
+            markdowner = Markdown(safe_mode='escape', extras=['strike', 'target-blank-links'])
+            atags = re.compile(r'<a ')
+            htags = re.compile(r'<(/?)h\d>')
+            tags = re.compile(r'^<p>(<code>)?|(</code>)?</p>$')
```

Comment:
Since `atags`, `htags` and `tags` are only used once below, there's no benefit of using `re.compile`, can just use the `re.sub` / `re.subn` functions directly.

Reply:
Are you sure about that?  They're used by the function `mark_to_html(text)`, which is called on (almost) every single line.